Parse/Public/Utilities/Conversion.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present, Parse, LLC.  All rights reserved.  This source code is licensed under the BSD-style license found in the LICENSE file in the root directory of this source tree.  An additional grant of patent rights can be found in the PATENTS file in the same directory.
+
+using Parse.Internal;
+using System;
+using System.Collections.Generic;
+
+namespace Parse.Utilities {
+  /// <summary>
+  /// A set of utilities for converting generic types between each other.
+  /// </summary>
+  public static class Conversion {
```

Comment:
I forgot the end result of our discussion. Do we want this to be `public` or just want to separate this out and mark as "pending decision whether we want to make it public/make it separate .dll or do nothing and leave it here"?